samples/ANCMStressTestSample/Startup.cs

```diff
@@ -0,0 +1,239 @@
+                await context.Response.WriteAsync("Path: " + context.Request.Path.Value + Environment.NewLine);
+                await context.Response.WriteAsync("Query: " + context.Request.QueryString.Value + Environment.NewLine);
+                await context.Response.WriteAsync(Environment.NewLine);
+
+                await context.Response.WriteAsync("Connection:" + Environment.NewLine);
+                await context.Response.WriteAsync("RemoteIp: " + context.Connection.RemoteIpAddress + Environment.NewLine);
+                await context.Response.WriteAsync("RemotePort: " + context.Connection.RemotePort + Environment.NewLine);
+                await context.Response.WriteAsync("LocalIp: " + context.Connection.LocalIpAddress + Environment.NewLine);
+                await context.Response.WriteAsync("LocalPort: " + context.Connection.LocalPort + Environment.NewLine);
+                await context.Response.WriteAsync(Environment.NewLine);
+
+                await context.Response.WriteAsync("Headers:" + Environment.NewLine);
+                foreach (var header in context.Request.Headers)
+                {
+                    await context.Response.WriteAsync(header.Key + ": " + header.Value + Environment.NewLine);
+                }
+                await context.Response.WriteAsync(Environment.NewLine);
+            });
+        }
+
+        private void WebSocket(IApplicationBuilder app)
+        {
+            app.UseWebSockets(new WebSocketOptions
+            {
+            });
+
+            app.Use(async (context, next) =>
+            {
+                if (context.WebSockets.IsWebSocketRequest)
+                {
+                    var webSocket = await context.WebSockets.AcceptWebSocketAsync();
+                    await Echo(webSocket);
+                }
+                else
+                {
+                    var wsScheme = context.Request.IsHttps ? "wss" : "ws";
+                    var wsUrl = $"{wsScheme}://{context.Request.Host.Host}:{context.Request.Host.Port}{context.Request.Path}";
+                    await context.Response.WriteAsync($"Ready to accept a WebSocket request at: {wsUrl}");
+                }
+            });
+        }
+
+        private async Task Echo(WebSocket webSocket)
+        {
+            var buffer = new byte[1024 * 4];
+            var result = await webSocket.ReceiveAsync(new ArraySegment<byte>(buffer), CancellationToken.None);
+            bool closeFromServer = false;
+            string closeFromServerCmd = "CloseFromServer";
+            int closeFromServerLength = closeFromServerCmd.Length;
+
+            bool echoBack = true;
+            int repeatCount = 1;
+
+            while (!result.CloseStatus.HasValue)
+            {
+                if ((result.Count == closeFromServerLength && System.Text.Encoding.ASCII.GetString(buffer).Substring(0, result.Count) == closeFromServerCmd)
```

Comment:
I don't feel like transmitting data across the websocket for commands is appropriate. I think these should be transmitted via Headers on the upgrade request test one individual component per websocket request.
cc @anurse